### PR TITLE
Enabling-more-agents-for-Batch-RL-and-cleanup

### PR DESCRIPTION
**Pull request by galleibo-intel**
allowing for the last training batch drawn to be smaller than batch_size + adding support for more agents in BatchRL by adding softmax with temperature to the corresponding heads + adding a CartPole_QR_DQN preset with a golden test + cleanups

The list of supported agents in BatchRL is now extended to - 
DQN
DDQN
Dueling DDQN
Categorical DQN (C51)
Rainbow DQN
QR DQN
Bootstrapped DQN
NEC
MMC
PAL

### CODE REVIEW

File: rl_coach/memories/episodic/episodic_experience_replay.py [hunk 1]
```diff
@@ -163,9 +163,8 @@ def get_shuffled_data_generator(self, size: int) -> List[Transition]:
         shuffled_transition_indices = list(range(self.last_training_set_transition_id))
         random.shuffle(shuffled_transition_indices)
 
-        # we deliberately drop some of the ending data which is left after dividing to batches of size `size`
-        # for i in range(math.ceil(len(shuffled_transition_indices) / size)):
-        for i in range(int(len(shuffled_transition_indices) / size)):
+        # The last batch drawn will usually be < batch_size (=the size variable)
+        for i in range(int(len(shuffled_transition_indices) / size) + 1):
```
Comment (nzmora) [inline rl_coach/memories/episodic/episodic_experience_replay.py]: If you are trying to round-up the loop-end condition then you have a bug here when 
```len(shuffled_transition_indices) / size == 0 ```
Why isn't the old code (in remark) good enough?
```
 range(math.ceil(len(shuffled_transition_indices) / size))
```

Comment (galleibo-intel) [reply rl_coach/memories/episodic/episodic_experience_replay.py]: Right, thanks. Will fix.

File: rl_coach/architectures/tensorflow_components/heads/dueling_q_head.py [hunk 2]
```diff
@@ -44,7 +44,12 @@ def _build_module(self, input_layer):
             self.action_advantage = self.action_advantage - self.action_mean
 
         # merge to state-action value function Q
-        self.output = tf.add(self.state_value, self.action_advantage, name='output')
+        self.q_values = self.output = tf.add(self.state_value, self.action_advantage, name='output')
+
+        # used in batch-rl to estimate a probablity distribution over actions
+        temperature = self.ap.network_wrappers[self.network_name].softmax_temperature
```
[the remark this answers is not in the document]
Comment (galleibo-intel) [reply rl_coach/architectures/tensorflow_components/heads/dueling_q_head.py]: Thanks, fixed.
